openstack/client.go

```diff
@@ -171,7 +171,7 @@ func v2auth(ctx context.Context, client *gophercloud.ProviderClient, endpoint st
 		tao := options
 		tao.AllowReauth = false
 		client.ReauthFunc = func() error {
-			err := v2auth(ctx, &tac, endpoint, tao, eo)
+			err := v2auth(context.Background(), &tac, endpoint, tao, eo)
```

Comment:
Perhaps you may use `client.Context` if it is not nil, then `context.Background()`.

Reply:
The provider client's context is already merged to the per-call context lower in the stack, when the actual HTTP call is delegated to the provider client itself.

Reply:
Aha, thanks for the explanation!